### ISSUE TITLE
Implement `group_by.()`?

### ISSUE THREAD

**Issue by markfairbanks** [markfairbanks/tidytable#375]
FYI @bnicenboim

**Comment by markfairbanks** [comment]
Closing this as #374 is no longer going to be implemented. I think we'll keep tidytable as-is.

**Comment by markfairbanks** [comment]
See #572

**Comment by markfairbanks** [comment]
@bnicenboim if you do give this a try let me know if you run into any issues. Naturally I have a bunch of internal tests to make sure things are working correctly, but there's always a chance I missed something.

There shouldn't be too many issues. The implementations are pretty straightforward. They just use the `.by` syntax in the background.

For example here's `filter.()` on a grouped tidytable:

``` r
filter..grouped_tt <- function(.df, ..., .by = NULL) {
  .by <- grouped_dot_by(.df, {{ .by }})
  out <- ungroup.(.df)
  out <- filter.(out, ..., .by = all_of(.by))
  group_by.(out, all_of(.by))
}
```

Basically they all ungroup --> use the normal function with `.by` --> regroup the output. And FYI regrouping is a very fast operation because there aren't any actual calculations occurring.
